### PR TITLE
W22 5pm 4 jz delete common

### PR DESCRIPTION
**Pull request by zjw9635468**
# Overview

Add delete button on the display page

# Issues Addressed

#19

# Details

Add delete button for every row. It is clickable and can successfully delete the common on that row.
Add tests to make coverage of the DisplayTable.js and AdminDisplayTablePage.js 100%

### CODE REVIEW

File: frontend/src/fixtures/commonsFixtures.js [hunk 1]
```diff
@@ -23,7 +29,18 @@ const commonsFixtures = {
...
+            "startingDate": "6/11/2021",
         }
+        //{
+        // "id": 1,
+        // "name": "Anika's Commons",
+        // "startingDate": 5,
+        // "endDate": "6/11/2021",
+        // "cowPrice": 50,
+        // "milkPrice": 15,
+        //}
```
Comment (pconrad) [inline frontend/src/fixtures/commonsFixtures.js]: Why is this commented out code still in the file?

File: frontend/src/main/components/Commons/DisplayTable.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            accessor: "milkPrice",
+        },
+        {
+            Header: "Start Date",
+            accessor: "startingDate",
+        },
+        // {
+        //     Header: "End Date",
+        //     accessor: "endDate",
+        // }
```
Comment (pconrad) [inline frontend/src/main/components/Commons/DisplayTable.js]: Why is this here?

File: frontend/src/main/components/Commons/EditCommonsForm.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      </Form.Group>
+ { /*
+  //     <Form.Group className="mb-3">
+  //       <Form.Label htmlFor="startingBalance">Starting Balance</Form.Label>
+  //       <Form.Control
+  //         id="startingBalance"
+  //         type="number"
+  //         step="0.01"
+  //         isInvalid={!!errors.startingBalance}
+  //         {...register("startingBalance", {
```
Comment (pconrad) [inline frontend/src/main/components/Commons/EditCommonsForm.js]: What is the reason for all of this commented out code?

File: frontend/src/tests/pages/AdminDisplayTablePage.test.js [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    //         "cowPrice": 15,
+    //     });
+
+    //     const { getByText, getByLabelText, getByTestId } = render(
+    //         <QueryClientProvider client={queryClient}>
+    //             <MemoryRouter>
+    //                 <AdminCreateCommonsPage />
+    //             </MemoryRouter>
+    //         </QueryClientProvider>
+    //     );
```
Comment (pconrad) [inline frontend/src/tests/pages/AdminDisplayTablePage.test.js]: What is the purpose of this commented out code?

File: frontend/src/tests/pages/HomePage.test.js [hunk 5]
```diff
@@ -70,9 +90,11 @@ describe("HomePage tests", () => {
         await waitFor(() => expect(getByTestId("commonsCard-button-Visit-1")).toBeInTheDocument());
         const visitButton = getByTestId("commonsCard-button-Visit-1");
         fireEvent.click(visitButton);
+        // await waitFor(() => expect(mockNavigate).toBeCalled);
+        // expect(mockNavigate).toBeCalledWith({ "to": "/play/1" });
```
Comment (pconrad) [inline frontend/src/tests/pages/HomePage.test.js]: Is this commented out code needed?

File: frontend/src/tests/utils/commonsUtils.test.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    //         const expectedToast =  `Edit Callback called on id: 17 name: Groundhog Day`
+
+    //         // act
+    //         const result = editCallback(cell);
+
+    //         // assert
+
+    //         expect(mockToast).toHaveBeenCalledWith(expectedToast);
+
+    //     });
```
Comment (pconrad) [inline frontend/src/tests/utils/commonsUtils.test.js]: Commented out code generally is not something we want to merge into the main branch.  

There are some exceptions, but not many.

**Review by pconrad — CHANGES_REQUESTED** [review]

File: frontend/src/fixtures/commonsFixtures.js (hunk 1 above)
Comment (zjw9635468) [reply frontend/src/fixtures/commonsFixtures.js]: deleted commented out lines

File: frontend/src/main/components/Commons/DisplayTable.js (hunk 2 above)
Comment (zjw9635468) [reply frontend/src/main/components/Commons/DisplayTable.js]: deleted commented out lines

File: frontend/src/main/components/Commons/EditCommonsForm.js (hunk 3 above)
Comment (zjw9635468) [reply frontend/src/main/components/Commons/EditCommonsForm.js]: deleted commented out lines

File: frontend/src/tests/pages/AdminDisplayTablePage.test.js (hunk 4 above)
Comment (zjw9635468) [reply frontend/src/tests/pages/AdminDisplayTablePage.test.js]: deleted commented out lines

File: frontend/src/tests/pages/HomePage.test.js (hunk 5 above)
Comment (zjw9635468) [reply frontend/src/tests/pages/HomePage.test.js]: deleted commented out lines

File: frontend/src/tests/utils/commonsUtils.test.js (hunk 6 above)
Comment (zjw9635468) [reply frontend/src/tests/utils/commonsUtils.test.js]: deleted commented out lines

**Review by kdeng23 — APPROVED** [review]
LGTM!!

**Review by samar1409 — DISMISSED** [review]
LGTM

**Review by kdeng23 — DISMISSED** [review]
The changes look good to me.

**Review by pconrad — DISMISSED** [review]
LGTM... will test on QA site after merging in PR #36 and merging main into this branch

**Review by samar1409 — APPROVED** [review]
LGTM

**Review by pconrad — APPROVED** [review]
LGTM.  Tested on Heroku
